php_phongo.c

```diff
@@ -1553,7 +1553,7 @@ static bool php_phongo_apply_options_to_uri(mongoc_uri_t* uri, bson_t* options)
 			!strcasecmp(key, MONGOC_URI_READPREFERENCE) ||
 			!strcasecmp(key, MONGOC_URI_READPREFERENCETAGS) ||
 			!strcasecmp(key, MONGOC_URI_SAFE) ||
-			!strcasecmp(key, MONGOC_URI_SLAVEOK) ||
+			!strcasecmp(key, MONGOC_URI_SECONDARYOK) ||
```

Comment:
If `slaveOk` handling was removed entirely from `php_phongo_apply_rp_options_to_uri`, what's the reason for renaming it to a new constant here?

This seems like it's related to [PHPC-1836](https://jira.mongodb.org/browse/PHPC-1836) and [CDRIVER-3989](https://jira.mongodb.org/browse/CDRIVER-3989) (removing `slaveOk` URI handling), but I don't see this constant in the corresponding libmongoc PR: https://github.com/mongodb/mongo-c-driver/pull/794

I found a reference to this in your [cdriver-3997](https://github.com/mongodb/mongo-c-driver/compare/master...alcaeus:cdriver-3997) branch, but it's still defined to "slaveok". Was there a more recent decision to continue parsing the legacy URI option and only rename the constant?

I didn't take a look at libmongoc's URI string parsing (not sure where to look as this work is split between so many tickets), but if PHP needs to explicitly ignore it here that suggests that libmongoc would otherwise still accept it by `mongoc_uri_option_is_bool` in the code down below.

What's the reason for libmongoc continuing to parse this if we're dropping it entirely in PHP? I realize they might not be able to drop the old constant due to ABI, but I expect they could stop using it and allow `slaveOk` in a URI string to start raising warnings for unsupported URI options (something we don't surface through PHP).

Reply:
Yes, this can be removed as libmongoc ignores the option completely